Support PhantomPinned as the trivial field

Currently only PhantomData is allowed.

```rust
use ref_cast::RefCast;
use std::marker::PhantomPinned;

#[derive(RefCast)]
#[repr(transparent)]
pub struct Struct {
    data: usize,
    #[trivial]
    marker: PhantomPinned,
}
```

```console
error[E0277]: the trait bound `PhantomPinned: ref_cast::trivial::Trivial` is not satisfied
 --> src/main.rs:9:13
  |
9 |     marker: PhantomPinned,
  |             ^^^^^^^^^^^^^ the trait `ref_cast::trivial::Trivial` is not implemented for `PhantomPinned`
  |
  = help: the following other types implement trait `ref_cast::trivial::Trivial`:
            ()
            PhantomData<T>
note: required by a bound in `assert_trivial`
 --> /home/david/.cargo/registry/src/github.com-1ecc6299db9ec823/ref-cast-1.0.9/src/trivial.rs:8:26
  |
8 | pub fn assert_trivial<T: Trivial>() {}
  |                          ^^^^^^^ required by this bound in `assert_trivial`
```

Workaround: `PhantomData<PhantomPinned>` does work and is equivalent.

```rust
#[derive(RefCast)]
#[repr(transparent)]
pub struct Struct {
    data: usize,
    marker: PhantomData<PhantomPinned>,
}
```

--- a/build.rs
+++ b/build.rs
@@ -8,6 +8,10 @@ fn main() {
         None => return,
     };
 
+    if minor < 33 {
+        println!("cargo:rustc-cfg=no_phantom_pinned");
+    }
+
     if minor < 38 {
         println!("cargo:rustc-cfg=no_intrinsic_type_name");
     }

--- a/derive/src/lib.rs
+++ b/derive/src/lib.rs
@@ -223,7 +223,10 @@ fn decide_trivial(fields: &Fields) -> Result<fn(&Field) -> Result<bool>> {
 fn is_implicit_trivial(field: &Field) -> Result<bool> {
     match &field.ty {
         Type::Tuple(ty) => Ok(ty.elems.is_empty()),
-        Type::Path(ty) => Ok(ty.path.segments.last().unwrap().ident == "PhantomData"),
+        Type::Path(ty) => {
+            let ident = &ty.path.segments.last().unwrap().ident;
+            Ok(ident == "PhantomData" || ident == "PhantomPinned")
+        }
         _ => Ok(false),
     }
 }

--- a/src/trivial.rs
+++ b/src/trivial.rs
@@ -1,8 +1,13 @@
 use core::marker::PhantomData;
+#[cfg(not(no_phantom_pinned))]
+use core::marker::PhantomPinned;
 
 pub trait Trivial {}
 
 impl Trivial for () {}
 impl<T: ?Sized> Trivial for PhantomData<T> {}
+
+#[cfg(not(no_phantom_pinned))]
+impl Trivial for PhantomPinned {}
 
 pub fn assert_trivial<T: Trivial>() {}

--- a/tests/ui/not-trivial.stderr
+++ b/tests/ui/not-trivial.stderr
@@ -7,6 +7,7 @@ error[E0277]: the trait bound `String: ref_cast::trivial::Trivial` is not satisf
   = help: the following other types implement trait `ref_cast::trivial::Trivial`:
             ()
             PhantomData<T>
+            PhantomPinned
 note: required by a bound in `assert_trivial`
  --> src/trivial.rs
   |

